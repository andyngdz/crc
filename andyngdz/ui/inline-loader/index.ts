export { InlineLoader } from './inline-loader';
export type { InlineLoaderProps } from './inline-loader';
