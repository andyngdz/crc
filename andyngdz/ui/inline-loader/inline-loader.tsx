import React from "react";
import { LinearProgress } from "@material-ui/core";

export type InlineLoaderProps = {
  loading: boolean;
};

export const InlineLoader: React.FC<InlineLoaderProps> = ({
  loading,
  children,
}) => {
  if (loading) {
    return <LinearProgress />;
  }

  return <>{children}</>;
};
