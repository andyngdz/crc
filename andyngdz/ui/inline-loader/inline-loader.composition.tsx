import React, { useEffect, useState } from "react";
import { Container, Box } from "@material-ui/core";
import { InlineLoader } from "./inline-loader";

export const BasicInlineLoader = () => {
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setTimeout(() => setLoading(false), 3000);
  }, []);

  return (
    <Container>
      <Box pt={4}>
        <InlineLoader loading={loading}>Hello, World!</InlineLoader>
      </Box>
    </Container>
  );
};
