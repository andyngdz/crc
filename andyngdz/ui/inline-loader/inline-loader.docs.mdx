---
labels: ['react', 'typescript', 'inline-loader']
description: 'A InlineLoader component.'
---

import { InlineLoader } from './inline-loader';

InlineLoader example:

A simple InlineLoader component with some text

Code Snippet:
```js
<InlineLoader text="hello from InlineLoader" />
```


Live Playground:
```js live
<InlineLoader text="hello from InlineLoader" />
```
