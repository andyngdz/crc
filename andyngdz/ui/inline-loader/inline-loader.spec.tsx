import React from "react";
import { render } from "@testing-library/react";
import { BasicInlineLoader } from "./inline-loader.composition";

describe("inline-loader", () => {
  it("should render with the correct text", () => {
    const { container } = render(<BasicInlineLoader />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
