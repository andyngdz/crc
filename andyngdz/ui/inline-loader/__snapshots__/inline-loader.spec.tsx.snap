// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline-loader should render with the correct text 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      />
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
      />
    </div>
  </div>
</div>
`;
